platforms/ide/problems/src/test/kotlin/org/gradle/problems/internal/impl/JsonWriterTest.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.problems.internal.impl
+
+import groovy.json.JsonSlurper
+import org.gradle.internal.cc.impl.problems.JsonWriter
+import org.gradle.internal.configuration.problems.StructuredMessage
+import org.gradle.internal.configuration.problems.writeStructuredMessage
+import org.gradle.internal.extensions.stdlib.uncheckedCast
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.hasEntry
+import org.junit.Test
+import java.io.StringWriter
+
+class JsonWriterTest {
+
+    @Test
+    fun `complex structure is valid JSON`() {
```

Comment:
It's not clear to me what do we test here. A better method name or more descriptive sample test data would be useful.